### PR TITLE
Add contributing section about Java repositories

### PR DESCRIPTION
**Pull request by sbesson**
Following the re-architecture of the Java components during Bio-Formats 5.3.0, a certain number of policies were adopted regarding the development, maintenance and distribution of the low-level Java repositories. This PR codifies these guidelines into the contributing documentation together with a table of the Java libraries following this process.

The top-level index page is also modified to remove the duplicate `..only: html` directive following the dropping of the PDF and uses the `toctree` directly to populate the contributing menu.

### CODE REVIEW

File: contributing/java-development.txt
Comment (mtbc) [inline contributing/java-development.txt]: maybe simply "monolithic"

File: contributing/java-development.txt
Comment (mtbc) [inline contributing/java-development.txt]: comply *with*

File: contributing/java-development.txt
Comment (mtbc) [inline contributing/java-development.txt]: hyphenate single-module

File: contributing/java-development.txt
Comment (mtbc) [inline contributing/java-development.txt]: can we link to these?

File: contributing/java-development.txt
Comment (mtbc) [inline contributing/java-development.txt]: this sentence seems to have gone a bit wrong ... maybe "following"

File: contributing/java-development.txt
Comment (mtbc) [inline contributing/java-development.txt]: maybe, *a* Maven component

File: contributing/java-development.txt
Comment (mtbc) [inline contributing/java-development.txt]: PGP-signed

File: contributing/java-development.txt
Comment (mtbc) [inline contributing/java-development.txt]: maybe instead of "with the usage of" something like "by using"

**Review by hflynn — APPROVED** [review]
All looks good to me. Does it need someone from the release team to give a thumbs up too before merging?

**Comment by hflynn** [comment]
Don't see any point blocking this any longer, going to merge and release, we can always edit later if someone spots something.

**Pull request merged**
